components/style/context.rs

```diff
@@ -120,17 +121,22 @@ pub struct TraversalStatistics {
     pub elements_matched: u32,
     /// The number of cache hits from the StyleSharingCache.
     pub styles_shared: u32,
+    /// Time spent in the traversal, in milliseconds.
+    pub traversal_time_ms: f64,
 }
 
 /// Implementation of Add to aggregate statistics across different threads.
 impl<'a> Add for &'a TraversalStatistics {
     type Output = TraversalStatistics;
     fn add(self, other: Self) -> TraversalStatistics {
+        debug_assert!(self.traversal_time_ms == 0.0 && other.traversal_time_ms == 0.0,
+                      "traversal_time_ms should be set at the end by the caller");
```

Comment:
Yeah, I just wanted it to show up in the same perf block for @shinglyu's automation, and this seemed like the simplest way to do it.

I agree it's a bit gross. The other way I considered was that each TraversalStatistics would capture a start time when instantiated, and then we'd take the min when we merge them, and rely on the fact that we must instantiate at least one ThreadLocalStyleContext (and therefore TraversalStatistics) very near to the beginning of the traversal. But that seemed like it required a lot of knowledge about the style system for people to verify that it was measuring the right thing, so I went with the dumb approach.